File: src/api/maestro/api/debug/IDebugRenderer.java
package maestro.api.debug;

/**
 * Interface for the debug rendering system.
 *
 * <p>Provides access to specialized renderers for different debug visualization needs.
 */
public interface IDebugRenderer {

    /**
     * Get the HUD debug renderer.
     *
     * @return the HUD debug renderer
     */
    IHudDebugRenderer getHudRenderer();
}

File: common/src/main/maestro/debug/IDebugRenderer.kt
package maestro.debug

/**
 * Interface for the debug rendering system.
 *
 * Provides access to specialized renderers for different debug visualization needs.
 */
interface IDebugRenderer {
    /**
     * Get the HUD debug renderer.
     *
     * @return the HUD debug renderer
     */
    fun getHudRenderer(): IHudDebugRenderer
}
